src/psbt.h

```diff
@@ -1231,6 +1231,9 @@ bool PSBTInputSignedAndVerified(const PartiallySignedTransaction psbt, unsigned
  **/
 bool SignPSBTInput(const SigningProvider& provider, PartiallySignedTransaction& psbt, int index, const PrecomputedTransactionData* txdata, int sighash = SIGHASH_ALL, SignatureData* out_sigdata = nullptr, bool finalize = true);
 
+/** Removes the unnecessary non_witness_utxos from a psbt. */
```

Comment:
I see that you’re just moving this code, but could you add a little more explanation here? When I was reviewing this, I was first trying to find out what `non_witness_utxos` and `witness_utxos` were, when we needed them, and finally learned per the previous PR when we could drop them. My understanding is that we can only drop `non_witness_utxos` when all inputs are segwit v1.

I would suspect that we might also be able to drop `witness_utxos` on non-segwit inputs when we do have the corresponding `non_witness_utxos`, and vice versa, drop the `non_witness_utxos` on segwit inputs, if we do have the `witness_utxos` for them. Is that accurate?

```suggestion
/** Reduces the size of the PSBT by dropping unnecessary `non_witness_utxos` (i.e. previous transactions) from a psbt when all inputs are segwit v1. */
```

Reply:
> I see that you’re just moving this code, but could you add a little more explanation here?

Done as per your suggestion above.
> I would suspect that we might also be able to drop witness_utxos on non-segwit inputs when we do have the corresponding non_witness_utxos,

I don't think `witness_utxo`s should ever get added to non-segwit inputs. 
> and vice versa, drop the non_witness_utxos on segwit inputs, if we do have the witness_utxos for them.

Currently I'm not sure, though in https://github.com/bitcoin/bitcoin/pull/25939#discussion_r1020917822 sipa said that in this situation all `non_witness_utxo`s should be kept. Even if this is not the case, changing this behavior would be out of the scope of this PR.